[release/6.0] ILVerify does not build in source-build

The .NET source-build tarball is failing to build on release/6.0.3xx with the following error:

```
/tarball/src/runtime.3a25a7f1cc446b60678ed25c9d829420d6321eba/artifacts/source-build/self/src/src/coreclr/tools/ILVerify/Program.cs(498,31): error CS0103: The name 'CommandHandler' does not exist in the current context [/tarball/src/runtime.3a25a7f1cc446b60678ed25c9d829420d6321eba/artifacts/source-build/self/src/src/coreclr/tools/ILVerify/ILVerify.csproj]
```

This stems from a couple of factors.  

1. Source build only builds one version of each repo.  The sdk's dependency on dotnet/command-line-api is the version included in source-build.
2. dotnet/command-line-api made some breaking changes and the dotnet/sdk took a [dependency on the latest release](https://github.com/dotnet/sdk/blob/release/6.0.3xx/eng/Version.Details.xml#L271) while dotnet/runtime is on an [older version](https://github.com/dotnet/runtime/blob/release/6.0/eng/Versions.props#L123).

Updating dotnet/runtime's version of command-line-api to the same as the sdk will solve the problem for release/6.0.3xx but it will then break release/6.0.2xx and release/6.0.1xx which are on the older version.  Upgrading the sdk to the latest command-line-api is a significant amount of work and was rejected to 6.0.2xx.

Options to fix this issue.
1. Exclude ILVerify from source-build.  I took a cursory look and see that ILVerification.nupkg is required but ILVerify might not be needed.
2. Update ILVerify to be conditioned so that it could build with either version (beta1 or beta2) of command-line-api.
3. Build both command-line-api versions in source-build.  This is a new feature which feels too for a servicing release.  It should only be a last resort after exhausting all other options.

cc @dotnet/source-build-internal

--- a/src/coreclr/.nuget/coreclr-packages.proj
+++ b/src/coreclr/.nuget/coreclr-packages.proj
@@ -1,8 +1,8 @@
 <Project Sdk="Microsoft.Build.Traversal" DefaultTargets="Pack">
   <ItemGroup Condition="'$(TargetOS)' == 'windows' or '$(DotNetBuildFromSource)' == 'true'">
     <ProjectReference Include="Microsoft.NET.Sdk.IL\Microsoft.NET.Sdk.IL.pkgproj" />
-    <ProjectReference Include="Microsoft.ILVerification\Microsoft.ILVerification.pkgproj" />
-    <ProjectReference Include="dotnet-ilverify\dotnet-ilverify.pkgproj" />
+    <ProjectReference Include="Microsoft.ILVerification\Microsoft.ILVerification.pkgproj" Condition="'$(DotNetBuildFromSource)' != 'true'" />
+    <ProjectReference Include="dotnet-ilverify\dotnet-ilverify.pkgproj" Condition="'$(DotNetBuildFromSource)' != 'true'" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(RuntimeFlavor)' == 'CoreCLR' and '$(DotNetBuildFromSource)' != 'true'">

